xcode-compat/src/main/kotlin/org/jetbrains/kotlin/xcodecompat/XcodeCompatPlugin.kt

```diff
@@ -28,10 +29,23 @@ open class KotlinXcodeExtension(private val project: Project) {
         val buildType = NativeBuildType.valueOf(System.getenv("CONFIGURATION")?.toUpperCase()
                 ?: "DEBUG")
         if (this.buildType == buildType) {
+            var dsymTask: Sync? = null
+
+            if (buildType == NativeBuildType.DEBUG) {
+                dsymTask = project.task<Sync>("buildForXcodeDSYM") {
+                    dependsOn(linkTask)
+                    val outputFile = linkTask.outputFile.get()
+                    val outputDSYM = File(outputFile.parent, outputFile.name + ".dSYM")
+                    from(outputDSYM)
+                    into(File(System.getenv("CONFIGURATION_BUILD_DIR"), outputDSYM.name))
+                }
+            }
+
             val buildForXcodeTask = project.task<Sync>("buildForXcode") {
-                dependsOn(linkTask)
-                from(linkTask.outputFile.get().parentFile)
-                into(System.getenv("CONFIGURATION_BUILD_DIR"))
+                dependsOn(dsymTask ?: linkTask)
```

Comment:
it looks fine to have a dependency from both tasks without a tricky selection. Then you may add the extra task dependency inside the first `if` statement

Reply:
But the shell script in Xcode project invokes only one task: the `buildForXcode`. The idea is that it also triggers `buildForXcodeDSYM`, if required.

Reply:
I see that now. I wonder why do we remove the dependency from `linkTask`. We might just add yet another dependency from the DSYM task instead, not touching the linkTask at all. Just a complexity warning.